components/board/genericlinux/board.go

```diff
@@ -86,43 +76,127 @@ func (b *Board) Reconfigure(
 	_ resource.Dependencies,
 	conf resource.Config,
 ) error {
-	newConf, err := b.convertConfig(conf)
+	newConfPtr, err := b.convertConfig(conf)
 	if err != nil {
 		return err
 	}
+	newConf := *newConfPtr
 
-	return b.ReconfigureParsedConfig(ctx, *newConf)
-}
-
-// ReconfigureParsedConfig is a public helper that should only be used
-// by the customlinux package.
-func (b *Board) ReconfigureParsedConfig(ctx context.Context, conf LinuxBoardConfig) error {
```

Comment:
This helper function had been split out during Natalie's intern project, but it's no longer needed, so I re-inlined it again.